### DISCUSSION TITLE
Progress on DataTable/DataGrid for WinUI 3

### DISCUSSION THREAD

**Discussion by Njoro007** [microsoft/microsoft-ui-xaml#10164] [General]
I am currently consulting on the progress of DataTable/DataGrid for WinUI 3. Coming from a WinForms background, I have extensive experience using DataGrid in WinForms applications. My primary concern is the implementation and functionality of DataTable in WinUI 3.

Could anyone provide updates or insights on the current state of DataTable/DataGrid in WinUI 3? Specifically, I am interested in:

Current capabilities and limitations

Best practices for migrating from WinForms DataGrid to WinUI 3 DataGrid

Any available resources or documentation that could assist in this transition

**Comment by haijialiu** [comment]
At present, The [DataGrid](https://learn.microsoft.com/en-us/windows/communitytoolkit/controls/datagrid) is not a component of winui 3, It's in  Windows Community Toolkit version 7.1.0. Well, [DataTable](https://github.com/CommunityToolkit/Labs-Windows/discussions/415) in Windows Community Toolkit version 8, but it's Experimental control.

**Comment by w-ahmad** [comment]
In their initial plan for version 1.7, they included a TableView control as an alternative to the datagrid. However, the first release for version 1.7 is now available, and the TableView control isn't there.

Take a look at the WinUI.TableView control available on https://github.com/w-ahmad/WinUI.TableView and see if it meets your requirements.
